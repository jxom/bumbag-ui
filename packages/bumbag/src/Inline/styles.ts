import { cssClass } from '../styled';
import { theme } from '../utils';

export const Inline = (styleProps) => cssClass`
  display: inline;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
