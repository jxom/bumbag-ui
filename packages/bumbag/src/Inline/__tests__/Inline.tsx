import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Inline } from '../Inline';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Inline>Hello world</Inline>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Inline position="absolute" top="0">
        Hello world
      </Inline>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Inline use="p">Hello world</Inline>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Inline props', () => {
      const { result } = renderHook(() => Inline.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Inline>{(InlineProps) => <div {...InlineProps}>Hello world</div>}</Inline>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Inline.base should render correctly', () => {
    const { container } = render(<Inline>hello world</Inline>, {
      theme: { Inline: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Inline>hello world</Inline>, {
      theme: { Inline: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
