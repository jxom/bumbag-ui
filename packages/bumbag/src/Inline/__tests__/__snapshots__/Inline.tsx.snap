// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Inline"
>
  Hello world
</p>
`;

exports[`composition hook should return with Inline props 1`] = `
Object {
  "className": "css-adelxu css-1pwrqjj css-1yuzjtp fp-Box fp-Inline",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Inline"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline;
}

<div
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Inline"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Inline"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline;
}

.emotion-2 {
  position: absolute !important;
  top: 0 !important;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Inline"
>
  Hello world
</div>
`;

exports[`theming Inline.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline;
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Inline"
>
  hello world
</div>
`;
